src/PowerShellEditorServices.Protocol/Server/LanguageServer.cs

```diff
@@ -726,6 +726,35 @@ RequestContext<object> requestContext
                     this.editorSession,
                     eventContext);
             }
+
+            // Convert the editor file glob patterns into an array for the Workspace
+            // TODO Is this thread-safe? this is hardly an atomic operation. But intermediate objects
+            // are costly. Given it's a config only change maybe a race condition won't occur?
+            editorSession.Workspace.ExcludeFilesGlob.Clear();
+            if (configChangeParams.Settings.Files != null && configChangeParams.Settings.Files.Exclude != null)
+            {
+                foreach(string pattern in configChangeParams.Settings.Files.Exclude.Keys)
+                {
+                    if (configChangeParams.Settings.Files.Exclude[pattern] && !editorSession.Workspace.ExcludeFilesGlob.Contains(pattern))
+                    {
+                        editorSession.Workspace.ExcludeFilesGlob.Add(pattern);
+                    }
+                }
+            }
+            if (configChangeParams.Settings.Search != null && configChangeParams.Settings.Search.Exclude != null)
+            {
+                foreach(string pattern in configChangeParams.Settings.Search.Exclude.Keys)
+                {
+                    if (configChangeParams.Settings.Search.Exclude[pattern] && !editorSession.Workspace.ExcludeFilesGlob.Contains(pattern))
```

Comment:
What do these key value pairs look like? I find it odd that you're doing a `foreach` for the keys and then using `configChangeParams.Settings.Search.Exclude[pattern]`.

You just cleared `editorSession.Workspace.ExcludeFilesGlob` on line 733 so we can assume it doesn't contain, right?

Reply:
Same as above, just different settings location.. Closing this comment as done.